### PR TITLE
doc: Document CursorMoved vim/neovim difference

### PR DESCRIPTION
**Pull request by ddickstein**
In Vim, the CursorMoved event is triggered either when the cursor is
moved in Normal or Visual mode or the text of the cursor line is
changed. In Neovim, the CursorMoved event is also triggered when the
cursor is moved to another window. This commit documents this nuance
in runtime/doc/vim_diff.txt.

### CODE REVIEW

**Comment by ddickstein** [comment]
Motivation for adding this documentation: https://github.com/ocaml/merlin/pull/1213

File: runtime/doc/vim_diff.txt [hunk 1]
```diff
@@ -151,6 +151,7 @@ Commands:
   |:sign-define| accepts a `numhl` argument, to highlight the line number
 
 Events:
+  |CursorMoved| also triggers when moving between windows
```
Comment (bfredl) [inline runtime/doc/vim_diff.txt:154]: This is a bit misleading, in vim it fires _most of the time_ when switching window, except when specifically the cursor position happens to match up perfectly.

Comment (zeertzjq) [reply runtime/doc/vim_diff.txt:154]:
```suggestion
  |CursorMoved| always triggers when moving between windows
```

Comment (zeertzjq) [reply runtime/doc/vim_diff.txt:154]: This should also be in the "Changed features" section.

**Comment by ddickstein** [comment]
I considered putting the comment under an "Auto-commands" section, but that seemed wrong since DirChanged is already  grouped under "Working directory."

**Comment by dundargoc** [comment]
Thank you for your contribution! We collect all typo fixes into a single pull request and merge it once it gets big enough: https://github.com/neovim/neovim/pull/17250

**Pull request closed without merging**